src/tools/kata-ctl/src/arch/aarch64/mod.rs

```diff
@@ -39,4 +40,9 @@ mod arch_specific {
     pub fn get_checks() -> Option<&'static [CheckItem<'static>]> {
         Some(CHECK_LIST)
     }
+
+    // Guest protection is not supported on ARM64.
+    pub fn available_guest_protection() -> &'static str{
```

Comment:
I think it would be better to return a `Result` here to allow the caller (and the tests) to distinguish between the following scenarios:

- User doesn't have sufficient privileges to determine if guest protection is available.
- Guest protection check failed (on a system that does have guest protection).

Something like:

```rust
use thiserror::Error;

#[derive(Debug)]
pub enum GuestProtection {
    NoProtection,  // [1]
    TDX,
    SGX,
    // ...
}

#[derive(Error, Debug)]
pub enum Error {
    #[error("No permission to check guest protection")]
    GuestProtectNoPerms,

    #[error("Failed to check guest protection: {0}")]
    GuestProtectCheckFailed(String),

    #[error("Invalid guest protection value: {0}")]
    GuestProtectInvalidValue(String),
}

pub fn available_guest_protection() -> Result<GuestProtection> {
    // If EPERM, return Err(Error::GuestProtectNoPerms).
    // If no protection, return Ok(GuestProtection::NoProtection).
    // If check fails, return Err(Error::GuestProtectCheckFailed(some_value.to_string())).
    // If protection available, return something like, Ok(GuestProtection::TDX).
}
```

[1] - Note that an alternative to definining `GuestProtection::NoProtection` would be to make `available_guest_protection()` return `Result<Option<GuestProtection>>` where the `Option` part would be taken to mean, "does the user have permission to check"?  `None` would equate to "no" and `Some(x)` would be "yes":

| user has perms to check protection | protection available? | test worked? | return value |
|-|-|-|-|
| no | no | n/a | `Err(Error::GuestProtectNoPerms)` |
| no | yes  (but user won't know that)| n/a | `Err(Error::GuestProtectNoPerms)` |
| yes | no | yes | `Ok(None)` |
| yes | yes | yes | `Ok(Some(GuestProtection::TDX)` |
| yes | yes | no | `Err(Error::GuestProtectCheckFailed(some_value.to_string()))` |